File: PDP/Odev2/KISIM2/src/Rastgele.c
/**
*
* @author Melih Ensar BARIŞIK ensar.barisik@ogr.sakarya.edu.tr / Ömer Can ÇALIŞIR omer.calisir@ogr.sakarya.edu.tr
* @since  25.04.2020
* <p>
*  Rastgele sayı üretme kısmı 
* </p>
*/
#include "Rastgele.h"
#include<stdio.h>
#include<stdlib.h>
#include <time.h>


// nesne yönelimli benzetme için sözde kurucu fonksiyonu
Rastgele RastgeleOlustur()
{
	Rastgele this;
	// struct için bellekte yer ayırma işlemi
	this=(Rastgele)malloc(sizeof(struct RASTGELE));
	
	// saniye çekme işlemi
	this->next=time(NULL);
	
	// fonksiyon göstericilerine oluşturulan
    // fonksiyonların adreslerinin verilmesi işlemi
	this->randomRakamUret=&_randomRakamUret;
	this->yokediciRastgele=&_yokediciRastgele;
	
	return this;
}


// istenilen aralıkta random bir sayi üreten fonksiyon
long int _randomRakamUret(const Rastgele this,long int ilk,long int son){

	this->next = this->next * 1103515245 + 12345;
	long int fark = son-ilk;
	// üretilen sayının istenilen aralığın dışında olmaması için
	// fark ile modunun alınması ve alt sınırla toplanması işlemi
	long int mod = (unsigned int)(this->next/65536)%fark;
	long int randomSayi = ilk + mod;
	return randomSayi;
}

// bellekten alınan yerin geri teslimi için yıkıcı fonksiyon
void _yokediciRastgele(Rastgele this)
{
	if(this==NULL) return;
	free(this);
}

File: PDP/Odev2/KISIM1/Odev2Kutuphane/src/RandomPacket/Rastgele.java
/**
*
* @author Melih Ensar BARIŞIK ensar.barisik@ogr.sakarya.edu.tr / Ömer Can ÇALIŞIR omer.calisir@ogr.sakarya.edu.tr
* @since  25.04.2020
* <p>
*  Rastgele sayı üretme kısmı 
* </p>
*/
package RandomPacket;

/**
 *
 * @author bar_i
 */
public class Rastgele {
    
    //Sıfırdan, aldığı limit parametresinin değerine kadar bir rastgele sayı üretir.
     public static int produceRandomDigit(int limit)
    {
        
     long preserver;
     //Rastgele sayı elde etmek için sistemden nano time ı alıp yüze bölüyoruz.
     preserver = System.nanoTime()/100;
     
     return (int)(preserver%(limit));
    }    
    
     //İstenen basamakta sayı üreten fonksiyon
    public static String produceRandomNumber(int decimalPlaces) 
    {
       String number ="";
       
       for(int i =0; i<decimalPlaces; i++)
       {
           //Her bir basamak için produceRandomDigit kullanarak rakam alır ve bu rakamaları birleştirir.
           number = number + produceRandomDigit(9);
       }
     
        return number;
    }
    
    
}
